modules/miscutil/lib/recommender.py

```diff
@@ -0,0 +1,138 @@
+
+import json
+
+from invenio.bibfield import get_record
+from invenio.config import CFG_BASE_URL, \
+    CFG_RECOMMENDER_REDIS, \
+    CFG_RECOMMENDER_PREFIX, \
+    CFG_SITE_RECORD
+from invenio.webuser import collect_user_info
+
+
+_RECOMMENDATION_REDIS = None
+
+
+def get_recommended_records(recid, user_id, maximum=3):
+    """
+    Record recommendations.
+
+    @param recid: Record id for the recommendation.
+    @param threshold: Not used but there for compatibility.
+    @return: List of recommended records [{
+                                          'number': ,
+                                          'record_url': ,
+                                          'record_title': ,
+                                          'record_authors': ,
+                                          }, ]
+    """
+    global _RECOMMENDATION_REDIS
+
+    if CFG_RECOMMENDER_REDIS == "":
+        # Recommender is not configured.
+        return []
+
+    if _RECOMMENDATION_REDIS is None:
+        import redis
+        _RECOMMENDATION_REDIS = redis.StrictRedis(host=CFG_RECOMMENDER_REDIS,
+                                                  port=6379,
+                                                  db=0)
+    # Get recommendation
+    key = "{0}{1}".format(CFG_RECOMMENDER_PREFIX, recid)
+    recommendations = _RECOMMENDATION_REDIS.get(key)
+    if not recommendations:
+        # No recommendations available.
+        return []
+
+    recommendations = json.loads(recommendations)
+
+    records = recommendations.get('records')
+    recommender_version = recommendations.get('version', 0)
+    return _format_recommendation(records, recid, recommender_version,
+                                  user_id, maximum)
+
+
+def _format_recommendation(recommended_records, recid_source, recommender_sys,
+                           user_id=0, maximum=3):
+    """
+    Load the record information for each record in a list.
+
+    @param recommended_records: List of records [1234 ,344 ,342]
+    @param recid_source: Record id where the recommendations are displayed.
+    @param recommender_sys: Recommender System id. For custom event.
+    @param maximum: Maximum suggestions to show
+    @return: List of recommended records [{
+                                          'number': ,
+                                          'record_url': ,
+                                          'record_title': ,
+                                          'record_authors': ,
+                                          }, ]
+    """
+    from invenio.webstat import get_url_customevent
+    from invenio.search_engine import check_user_can_view_record
+
+    check_same_title = []
+    suggestions = []
+    rec_count = 1
+    for recid in recommended_records:
+        try:
+            # check if user can view record
+            user_info = collect_user_info(user_id)
+            # TODO: maybe search_engine.py(393) - record_public_p(recid) better
```

Comment:
:+1: just showing public records should be enough, otherwise it gets more complex (for eg, restricted records that only X ppl  (where X < 5) have rights to see)